Java2020/src/lesson201130/ArraysClassUsage.java

```diff
@@ -29,8 +29,9 @@ public static void main(String[] args) {
 	}
 
 	private static int[] insert(int[] a, int key, int index) {
-		// TODO this is your home work
-		return null;
+		int[] aux = Arrays.copyOf(a, a.length + a.length >> 1);
+		aux[index] = key;
+		return aux;
```

Comment:
this will not work.  aux[index] contains some element, you'll just replace it with key.
you should shift everything starting from 'index' to the right.